overlord/snapstate/autorefresh.go

```diff
@@ -254,54 +256,74 @@ func (m *autoRefresh) Ensure() error {
 		logger.Debugf("Next refresh scheduled for %s.", m.nextRefresh.Format(time.RFC3339))
 	}
 
-	// should we hold back refreshes?
-	holdTime, err := m.EffectiveRefreshHold()
+	held, holdTime, err := m.isRefreshHeld(refreshSchedule)
 	if err != nil {
 		return err
 	}
-	if holdTime.After(now) {
-		return nil
-	}
-	if !holdTime.IsZero() {
-		// expired hold case
-		m.clearRefreshHold()
-		if m.nextRefresh.Before(holdTime) {
-			// next refresh is obsolete, compute the next one
-			delta := timeutil.Next(refreshSchedule, holdTime, maxPostponement)
-			now = time.Now()
-			m.nextRefresh = now.Add(delta)
-		}
-	}
 
 	// do refresh attempt (if needed)
-	if !m.nextRefresh.After(now) {
-		var can bool
-		can, err = m.canRefreshRespectingMetered(now, lastRefresh)
-		if err != nil {
-			return err
-		}
-		if !can {
-			// clear nextRefresh so that another refresh time is calculated
-			m.nextRefresh = time.Time{}
-			return nil
+	if !held {
+		// expired hold case
+		if !holdTime.IsZero() {
```

Comment:
we're not too consistent about that in the codebase, but let's at least try:

```suggestion
		if !holdTime.IsZero() {
			// expired hold case		
```